protected/humhub/modules/file/models/File.php

```diff
@@ -189,12 +189,29 @@ public function canRead($userId = "")
     }
 
     /**
-     * Checks if given file can deleted.
+     * Checks if given file can be deleted.
      *
      * If the file is not an instance of ContentActiveRecord or ContentAddonActiveRecord
      * the file is readable for all unless there is method canEdit or canDelete implemented.
+     *
+     * @param int $userId
+     * @return bool
      */
     public function canDelete($userId = null)
+    {
+        return $this->canEdit($userId);
```

Comment:
Why return `canEdit()` if it's still returning `getPolymorphicRelation()` regardless, I'm also curious to why `canEdit()` should be used for `canDelete()` at all? Just slightly confused and would like validation before trying out some tests😄

Reply:
I only ask because `canEdit()` and `canDelete()` should do completely different things, `canEdit()` should be a checker if a file can be edited, while `canDelete()` should be a checker if the file can be deleted, with how it's implemented here it just confuses me looking at the code itself 😅

Reply:
@ArchBlood Thanks for the review, please note I just made a new method `File::canEdit()` as copy of the already existing method `File::canDelete()` in order to keep better call like `if ($file->canEdit()) { ... }` from controller code for example.
I.e. yes these 2 methods have same code, but I am not sure we have a separate permissions: one for edit and one for delete, usually if user can edit/update/modify an object then such user can delete the object.

Probably we should use another name like `canManage` or `canModify`, so code will be like:

```
function canManage()
{
    // All generic code here
}

function canEdit()
{
    return $this->canManage();
}

function canDelete()
{
    return $this->canManage();
}
```